assets/js/modules/analytics-4/components/widgets/TopCitiesWidget.stories.js

```diff
@@ -0,0 +1,146 @@
+ */
+import { CORE_USER } from '../../../../googlesitekit/datastore/user/constants';
+import { MODULES_ANALYTICS_4 } from '../../datastore/constants';
+import {
+	provideKeyMetrics,
+	provideModules,
+} from '../../../../../../tests/js/utils';
+import { withWidgetComponentProps } from '../../../../googlesitekit/widgets/util';
+import { getAnalytics4MockResponse } from '../../utils/data-mock';
+import { replaceValuesInAnalytics4ReportWithZeroData } from '../../../../../../.storybook/utils/zeroReports';
+import WithRegistrySetup from '../../../../../../tests/js/WithRegistrySetup';
+import TopCitiesWidget from './TopCitiesWidget';
+
+const reportOptions = {
+	startDate: '2020-08-11',
+	endDate: '2020-09-07',
+	dimensions: [ 'city' ],
+	metrics: [ { name: 'totalUsers' } ],
+	orderby: [
+		{
+			metric: {
+				metricName: 'totalUsers',
+			},
+			desc: true,
+		},
+	],
+	limit: 3,
+};
+
+const WidgetWithComponentProps =
+	withWidgetComponentProps( 'test' )( TopCitiesWidget );
+
+const Template = ( { setupRegistry, ...args } ) => (
+	<WithRegistrySetup func={ setupRegistry }>
+		<WidgetWithComponentProps { ...args } />
+	</WithRegistrySetup>
+);
+
+export const Ready = Template.bind( {} );
+Ready.storyName = 'Ready';
+Ready.args = {
+	setupRegistry: ( registry ) => {
+		const report = getAnalytics4MockResponse( reportOptions );
+		// Calculate sum of metricValues for all rows
+		const rowsSum = report.rows.reduce( ( total, row ) => {
+			return total + Number( row.metricValues[ 0 ].value );
+		}, 0 );
+
+		// Generate totalValueForAllCities that is higher than the sum
+		const totalValueForAllCities = rowsSum * 2;
+
+		// Adjust totals field in the mock response
+		report.totals = [
+			{
+				dimensionValues: [ { value: 'RESERVED_TOTAL' } ],
+				metricValues: [ { value: totalValueForAllCities.toString() } ],
+			},
+		];
+		registry.dispatch( MODULES_ANALYTICS_4 ).receiveGetReport( report, {
+			options: reportOptions,
+		} );
+	},
+};
+Ready.scenario = {
+	label: 'KeyMetrics/TopCitiesWidget/Ready',
+	delay: 250,
+};
+
+export const Loading = Template.bind( {} );
+Loading.storyName = 'Loading';
+Loading.args = {
+	setupRegistry: ( { dispatch } ) => {
+		dispatch( MODULES_ANALYTICS_4 ).startResolution( 'getReport', [
+			reportOptions,
+		] );
+	},
+};
+
+export const ZeroData = Template.bind( {} );
+ZeroData.storyName = 'Zero Data';
+ZeroData.args = {
+	setupRegistry: ( { dispatch } ) => {
+		const report = getAnalytics4MockResponse( reportOptions );
+		const zeroReport =
+			replaceValuesInAnalytics4ReportWithZeroData( report );
+
+		dispatch( MODULES_ANALYTICS_4 ).receiveGetReport( zeroReport, {
+			options: reportOptions,
+		} );
+	},
+};
+ZeroData.scenario = {
+	label: 'KeyMetrics/TopCitiesWidget/ZeroData',
+	delay: 250,
+};
+
+export default {
+	title: 'Key Metrics/TopCitiesWidget',
+	decorators: [
+		( Story, { args } ) => {
+			const setupRegistry = ( registry ) => {
+				global._googlesitekitUserData.isUserInputCompleted = false;
```

Comment:
Instead of using `global`, could we use the `core/user` `receiveIsUserInputCompleted` action?

Reply:
Updated ✔️